bottlecap/src/traces/trace_aggregator.rs

```diff
@@ -7,11 +7,24 @@ use std::collections::VecDeque;
 /// <https://github.com/DataDog/datadog-agent/blob/9d57c10a9eeb3916e661d35dbd23c6e36395a99d/pkg/trace/writer/trace.go#L27-L31>
 pub const MAX_CONTENT_SIZE_BYTES: usize = (32 * 1_024 * 1_024) / 10;
 
+// Bundle SendDataBuilder with payload size because SendDataBuilder doesn't
+// expose a getter for the size
+pub struct SendDataBuilderInfo {
+    pub builder: SendDataBuilder,
+    pub size: usize,
+}
+
+impl SendDataBuilderInfo {
+    pub fn new(builder: SendDataBuilder, size: usize) -> Self {
+        Self { builder, size }
+    }
+}
+
```

Comment:
Adding `SendDataBuilderInfo` in `trace_aggregator.rs` because this is where the `size` is consumed. Let me know if you have a better place.